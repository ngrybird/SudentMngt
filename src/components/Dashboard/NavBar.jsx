import React from "react";
import profileAvatar from "../../images/avatar-header.png";
import vector from "../../images/Vector.png";

class NavBar extends React.Component {
  render() {
    return (
      <div className="navbar">
        <div className="profile">
          <span className="profile-info">
            <div className="profile-name" data-testid="id-profile-name">{this.props.name}</div>
            <div className="profile-designation">{this.props.position}</div>
          </span>
          <span className="profile-avatar">
            <img src={profileAvatar} height="50px" width="50px" alt="avatar" />
          </span>
          <span className="profile-more">
            <img src={vector} height="6px" width="12px" alt="avatar" />
          </span>
        </div>
      </div>
    );
  }
}
export default NavBar;
