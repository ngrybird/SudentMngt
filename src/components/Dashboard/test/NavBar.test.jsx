import React from "react";
import { render } from "@testing-library/react";
import NavBar from "../NavBar";

describe("<NavBar/>", () => {
  const profile = {
    name: "Cody Simmons",
    position: "Lecturer"
  };
  it("Verify Name on the profile", () => {
    const { container, getAllByTestId } = render(<NavBar {...profile} />);

    const profileName = getAllByTestId("id-profile-name");

    expect(profileName[0].innerHTML).toEqual("Cody Simmons");

    expect(container).toMatchSnapshot();
  });
});
