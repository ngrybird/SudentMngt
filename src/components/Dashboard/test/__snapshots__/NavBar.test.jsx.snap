// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> Verify Name on the profile 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="profile"
    >
      <span
        class="profile-info"
      >
        <div
          class="profile-name"
          data-testid="id-profile-name"
        >
          Cody Simmons
        </div>
        <div
          class="profile-designation"
        >
          Lecturer
        </div>
      </span>
      <span
        class="profile-avatar"
      >
        <img
          alt="avatar"
          height="50px"
          src="avatar-header.png"
          width="50px"
        />
      </span>
      <span
        class="profile-more"
      >
        <img
          alt="avatar"
          height="6px"
          src="Vector.png"
          width="12px"
        />
      </span>
    </div>
  </div>
</div>
`;
